TuviSRPLib/TuviSRPLibTests/ProtonSRPTests.cs

```diff
@@ -0,0 +1,107 @@
+﻿using Org.BouncyCastle.Crypto;
+using Org.BouncyCastle.Security;
+using Org.BouncyCastle.Utilities.Encoders;
+using System.Text;
+using TuviSRPLib;
+
+namespace TuviSRPLibTests
+{
+    internal class ProtonSRPTests
+    {
+        [Test]
+        public void FullCycleOfWorkTest()
+        {
+            var encodedN = "W2z5HBi8RvsfYzZTS7qBaUxxPhsfHJFZpu3Kd6s1JafNrCCH9rfvPLrfuqocxWPgWDH2R8neK7PkNvjxto9TStuY5z7jAzWRvFWN9cQhAKkdWgy0JY6ywVn22+HFpF4cYesHrqFIKUPDMSSIlWjBVmEJZ/MusD44ZT29xcPrOqeZvwtCffKtGAIjLYPZIEbZKnDM1Dm3q2K/xS5h+xdhjnndhsrkwm9U9oyA2wxzSXFL+pdfj2fOdRwuR5nW0J2NFrq3kJjkRmpO/Genq1UW+TEknIWAb6VzJJJA244K/H8cnSx2+nSNZO3bbo6Ys228ruV9A8m6DhxmS+bihN3ttQ==";
+            var decodedN = Base64.Decode(encodedN);
+            BigInteger N = new BigInteger(1, decodedN.Reverse().ToArray());
+            BigInteger g = new BigInteger("2");
+
+            string identity = "ivanov";
+            string password = "qwerty";
+            string salt = "some bytes";
+
+            Encoding enc = Encoding.UTF8;
+            byte[] identityBytes = enc.GetBytes(identity);
+            byte[] passwordBytes = enc.GetBytes(password);
+            byte[] saltBytes = enc.GetBytes(salt);
+
+            ProtonSRPClient client = new ProtonSRPClient();
+            ProtonSRPServer server = new ProtonSRPServer();
+            IDigest digest = new ExtendedHashDigest();
+
+            var verifier = ProtonSRPUtilities.CalculateVerifier(digest, N, g, saltBytes, identityBytes, passwordBytes);
+
+            server.Init(N, g, verifier, digest, new SecureRandom());
+            client.Init(N, g, digest, new SecureRandom());
+
+            BigInteger pubA = client.GenerateClientCredentials(saltBytes, identityBytes, passwordBytes);
+            BigInteger pubB = server.GenerateServerCredentials();
+
+            BigInteger serverSecret = server.CalculateSecret(pubA);
+            BigInteger clientSecret = client.CalculateSecret(pubB);
+
+            BigInteger M1 = client.CalculateClientEvidenceMessage();
+            Assert.IsTrue(server.VerifyClientEvidenceMessage(M1), "Message M1 is not verified.");
+
+            BigInteger M2 = server.CalculateServerEvidenceMessage();
+
+            Assert.IsTrue(client.VerifyServerEvidenceMessage(M2), "Message M2 is not verified.");
+
+            BigInteger clientKey = client.CalculateSessionKey();
+            BigInteger serverKey = server.CalculateSessionKey();
+
+            Assert.AreEqual(clientKey, serverKey);
+        }
+
+        [Test]
+        public void ClientSideWorkTest()
```

Comment:
I would added a negative test, just to be sure that authentication fails if the client provides incorrect data